repositories/settings/core.go

```diff
@@ -22,3 +37,46 @@ func (c *core) setResolution(width, height int32) {
 func (c *core) getResolution() (int32, int32) {
 	return c.width, c.height
 }
+
+func (c *core) setFromReader(reader io.Reader) error {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		err := scanner.Err()
+		if err != nil {
+			return err
+		}
+		if strings.Trim(line, " \t") == "" {
+			continue
+		}
+		elements := strings.Split(line, "=")
+		if len(elements) != 2 {
+			return errors.New("malformed settings line: expected key=value")
```

Comment:
This should be a sentinel error.
This error is not tested.

Reply:
This has been updated.

Reply:
I don't think we should be using sentinel errors. I was convinced by this article: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully